core/src/com/po/fuck/model/weapons/Gun.java

```diff
@@ -25,9 +29,17 @@ public boolean attack() {
             return false;
 
         Vector2 direction = getDirection(), gunPosition = getPosition();
-
         gunPosition.add(new Vector2(geometryData.getWidth() / 2, 0).rotateRad(direction.angleRad()));
-        Manager.create(shoot(gunPosition, getDirection()));
+
+        WeaponSpriteInfo info = WeaponSpriteManager.getWeaponSpriteInfo(this.getClass());
+        Vector2 realMuzzlePosition = info.muzzlePosition.cpy();
```

Comment:
Stop getting it here each time, it will not change, move into class variables with `protected`!